### PR TITLE
Alternative encryption

### PR DESCRIPTION
**Pull request by ilanolkies**
Add alternative encryption algorithm hashing r and s of signature for web3 providers with no wallet-side encryption support

### CODE REVIEW

**Review by jessgusclark — APPROVED** [review]
Looks good.

File: modules/ipfs-cpinner-client/src/encryption-manager/aes.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  key: hmac.substring(0, 64),
+  macKey: hmac.substring(64, 128)
+}))
+
+export const encrypt = (key, message, macKey) => {
+  const iv = encHex.stringify(WordArray.random(16))
+
+  const c = AES.encrypt(message, key, { iv: iv, mode: ModeCTR }).toString()
+  // do a MAC preventing padding oracle attacks
+  const m = HmacSHA256(iv + c, macKey).toString()
```
Comment (luisfontes19) [inline modules/ipfs-cpinner-client/src/encryption-manager/aes.ts:36]: I just noticed that iv is expected to be a WordArray and not a string. Still it seems to have the same behaviour both ways. If you want to you can change it, but seems to be ok. (Same for decrypting)

File: modules/ipfs-cpinner-client/src/encryption-manager/with-signer.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  private macKey: string
+
+  constructor (key: string, macKey: string) {
+    this.key = key
+    this.macKey = macKey
+  }
+
+  encrypt = (data: string) => {
+    const cipher = encrypt(this.key, data, this.macKey)
+    return Promise.resolve(Buffer.from(cipher, 'utf-8').toString('hex'))
```
Comment (luisfontes19) [inline modules/ipfs-cpinner-client/src/encryption-manager/with-signer.ts:16]: cipher should be already an hex string so no need to convert it again

```suggestion
    return Promise.resolve(cipher)
```

File: modules/ipfs-cpinner-client/src/encryption-manager/with-signer.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  }
+
+  decrypt = (data: string) => {
+    const cipher = Buffer.from(data, 'hex').toString('utf-8')
+    return Promise.resolve(decrypt(this.key, cipher, this.macKey))
+  }
```
Comment (luisfontes19) [inline modules/ipfs-cpinner-client/src/encryption-manager/with-signer.ts:22]: data should already be an hex string so no need to convert it again

```suggestion
  decrypt = (cipher: string) => {
    return Promise.resolve(decrypt(this.key, cipher, this.macKey))
  }
```

**Review by luisfontes19 — APPROVED** [review]
LGTM

**Review by luisfontes19 — APPROVED** [review]

**Pull request merged**